datumaro/plugins/coco_format/extractor.py

```diff
@@ -124,6 +147,42 @@ def _load_items(self, loader):
 
         return items
 
+    def _load_panoptic_items(self, config, panoptic_images):
+        import numpy as np
+        items = OrderedDict()
+
+        def bgr2id(color):
+            return color[:, :, 2] + 256 * color[:, :, 1] + 256 * 256 * color[:, :, 0]
+
+        imgs_info = {}
+        for img in config['images']:
+            imgs_info[img['id']] = img
+
+        for ann in config['annotations']:
+            img_id = int(ann['image_id'])
+            image_path = osp.join(self._images_dir, imgs_info[img_id]['file_name'])
+            image_size = (imgs_info[img_id].get('height'), imgs_info[img_id].get('width'))
+            image = Image(path=image_path, size=image_size)
+            anns = list()
+
+            panoptic_image_path = osp.join(panoptic_images, ann['file_name'])
+            panoptic_format = np.array(load_image(panoptic_image_path, dtype=np.uint32))
+            pan = bgr2id(panoptic_format)
+            for segm_info in ann['segments_info']:
+                cat_id = segm_info['category_id']
+                segm_id = segm_info['id']
+                mask = pan == segm_id
+                rle = mask_utils.encode(np.asfortranarray(mask.astype('uint8')))
```

Comment:
Please, update the implementation of `datumaro.util.mask_tools.mask_to_rle` if this code works faster. This can be done later.

Reply:
Yes, will be done in separate PR